applications/CompressiblePotentialFlowApplication/python_scripts/stochastic_potential_flow_response.py

```diff
@@ -0,0 +1,332 @@
+        else:
+            primal_parameters.AddString("adjoint_parameters_path", self.response_settings["adjoint_settings"].GetString())
+        if primal_parameters.Has("design_surface_sub_model_part_name"):
+            primal_parameters["design_surface_sub_model_part_name"].SetString(self.design_surface_sub_model_part_name)
+        else:
+            primal_parameters.AddString("design_surface_sub_model_part_name", self.design_surface_sub_model_part_name)
+        open(self.response_settings["primal_settings"].GetString(), 'w').write(primal_parameters.PrettyPrintJsonString())
+
+        self.primal_model_part = _GetModelPart(model, primal_parameters["solver_settings"])
+
+        self.primal_analysis = potential_flow_analysis.PotentialFlowAnalysis(model, primal_parameters)
+
+        self.primal_data_transfer_with_python = self.response_settings["primal_data_transfer_with_python"].GetBool()
+
+        # Create the adjoint solver
+        adjoint_parameters = _CheckParameters(self._GetAdjointParameters())
+        adjoint_model = KratosMultiphysics.Model()
+        self.adjoint_model_part = _GetModelPart(adjoint_model, adjoint_parameters["solver_settings"])
+
+        self.adjoint_analysis = potential_flow_analysis.PotentialFlowAnalysis(adjoint_model, adjoint_parameters)
+
+        self.primal_state_variables = [KCPFApp.VELOCITY_POTENTIAL, KCPFApp.AUXILIARY_VELOCITY_POTENTIAL]
+
+    def Initialize(self):
+        self.primal_analysis.Initialize()
+        self.adjoint_analysis.Initialize()
+
+    def InitializeSolutionStep(self):
+        self.primal_model_part.RemoveSubModelPart("fluid_computational_model_part")
+        KratosMultiphysics.ModelPartIO(self.auxiliary_mdpa_path, KratosMultiphysics.IO.WRITE | KratosMultiphysics.IO.MESH_ONLY).WriteModelPart( self.primal_model_part)
+
+        self._RunXMC()
+
+        # save lift coefficient
+        for qoi_counter in range (0,1):
+            for index in range (len(self.xmc_analysis.monteCarloSampler.indices)):
+                self.xmc_analysis.monteCarloSampler.indices[index].qoiEstimator[qoi_counter] = get_value_from_remote(self.xmc_analysis.monteCarloSampler.indices[index].qoiEstimator[qoi_counter])
+                sample_counter = self.xmc_analysis.monteCarloSampler.indices[index].qoiEstimator[qoi_counter]._sampleCounter
+                S1 = float(get_value_from_remote(self.xmc_analysis.monteCarloSampler.indices[index].qoiEstimator[qoi_counter].powerSums[0][0]))
+                S2 = float(get_value_from_remote(self.xmc_analysis.monteCarloSampler.indices[index].qoiEstimator[qoi_counter].powerSums[1][0]))
+                h1 = float(get_value_from_remote(mdccm.computeCentralMomentsOrderOneDimensionZero(S1,sample_counter)))
+                h2 = float(get_value_from_remote(mdccm.computeCentralMomentsOrderTwoDimensionZero(S1,S2,sample_counter)))
+                self._value = h1
```

Comment:
This procedure is very specific for MC algorithms. For MLMC, where `len(self.xmc_analysis.monteCarloSampler.indices)>1`, you would be overriding the expected value (h1) value. If you are interested in the exp value estimation, I would do something as 

```suggestion
            h1_values = []
            for index in range (len(self.xmc_analysis.monteCarloSampler.indices)):
                self.xmc_analysis.monteCarloSampler.indices[index].qoiEstimator[qoi_counter] = get_value_from_remote(self.xmc_analysis.monteCarloSampler.indices[index].qoiEstimator[qoi_counter])
                h1_values.append(float(get_value_from_remote(self.xmc_analysis.monteCarloSampler.indices[index].qoiEstimator[qoi_counter].value(order=1))))
            self._value = sum(h1_values)
```

Reply:
Let me double check what I wrote.

Reply:
The point is that for each index/level you have an expected value estimation. I have tried on another machine to use the `value` method, but it seems there is some issue.